src/crimson/osd/main.cc

```diff
@@ -238,6 +240,13 @@ int main(int argc, const char* argv[])
           local_conf().parse_config_files(conf_file_list).get();
           local_conf().parse_env().get();
           local_conf().parse_argv(config_proxy_args).get();
+          if (auto log_file = local_conf()->log_file; !log_file.empty()) {
+            try {
+              logfile_set_ostream(log_file);
+            } catch (const std::system_error& e) {
+              std::cerr << "unable to set log file: " << e.what() << std::endl;
+            }
+         }
```

Comment:
```suggestion
          std::ofstream log_file;
          if (auto log_file = local_conf()->log_file; !log_file.empty()) {
            log_file.open(log_file_path, std::ios::app | std::ios::out);
            try {
              throw_system_error_on(!log_file);
            } catch (const std::system_error& e) {
              ceph_abort_msg(fmt::format("unable to open log file: {}", e.what()));
            }
            crimson::get_logger(ceph_subsys_osd).set_ostream(log_file);
          }
```
```suggestion
          if (auto log_file = local_conf()->log_file; !log_file.empty()) {
            try {
              logfile_set_ostream(log_file);
            } catch (const std::system_error& e) {
              std::cerr << "unable to set log file: " << e.what() << std::endl;
            }
         }
```
```suggestion
          if (auto log_file = local_conf()->log_file; !log_file.empty()) {
            try {
              logfile_set_ostream(log_file);
            } catch (const std::system_error& e) {
              std::cerr << "unable to set log file: " << e.what() << std::endl;
            }
         }
```
```suggestion
          if (auto log_file = local_conf()->log_file; !log_file.empty()) {
            try {
              logfile_set_ostream(log_file);
            } catch (const std::system_error& e) {
              std::cerr << "unable to set log file: " << e.what() << std::endl;
            }
         }
```
```suggestion
          if (auto log_file = local_conf()->log_file; !log_file.empty()) {
            try {
              logfile_set_ostream(log_file);
            } catch (const std::system_error& e) {
              std::cerr << "unable to set log file: " << e.what() << std::endl;
            }
         }
```
```suggestion
          if (auto log_file = local_conf()->log_file; !log_file.empty()) {
            try {
              logfile_set_ostream(log_file);
            } catch (const std::system_error& e) {
              std::cerr << "unable to set log file: " << e.what() << std::endl;
            }
         }
```
```suggestion
          if (auto log_file = local_conf()->log_file; !log_file.empty()) {
            try {
              logfile_set_ostream(log_file);
            } catch (const std::system_error& e) {
              std::cerr << "unable to set log file: " << e.what() << std::endl;
            }
         }
```
```suggestion
          if (auto log_file = local_conf()->log_file; !log_file.empty()) {
            try {
              logfile_set_ostream(log_file);
            } catch (const std::system_error& e) {
              std::cerr << "unable to set log file: " << e.what() << std::endl;
            }
         }
```
```suggestion
          if (auto log_file = local_conf()->log_file; !log_file.empty()) {
            try {
              logfile_set_ostream(log_file);
            } catch (const std::system_error& e) {
              std::cerr << "unable to set log file: " << e.what() << std::endl;
            }
         }
```
```suggestion
          if (auto log_file = local_conf()->log_file; !log_file.empty()) {
            try {
              logfile_set_ostream(log_file);
            } catch (const std::system_error& e) {
              std::cerr << "unable to set log file: " << e.what() << std::endl;
            }
         }
```
```suggestion
          if (auto log_file = local_conf()->log_file; !log_file.empty()) {
            try {
              logfile_set_ostream(log_file);
            } catch (const std::system_error& e) {
              std::cerr << "unable to set log file: " << e.what() << std::endl;
            }
         }
```
```suggestion
          if (auto log_file = local_conf()->log_file; !log_file.empty()) {
            try {
              logfile_set_ostream(log_file);
            } catch (const std::system_error& e) {
              std::cerr << "unable to set log file: " << e.what() << std::endl;
            }
         }
```
```suggestion
          if (auto log_file = local_conf()->log_file; !log_file.empty()) {
            try {
              logfile_set_ostream(log_file);
            } catch (const std::system_error& e) {
              std::cerr << "unable to set log file: " << e.what() << std::endl;
            }
         }
```
```suggestion
          if (auto log_file = local_conf()->log_file; !log_file.empty()) {
            try {
              logfile_set_ostream(log_file);
            } catch (const std::system_error& e) {
              std::cerr << "unable to set log file: " << e.what() << std::endl;
            }
         }
```
```suggestion
          if (auto log_file = local_conf()->log_file; !log_file.empty()) {
            try {
              logfile_set_ostream(log_file);
            } catch (const std::system_error& e) {
              std::cerr << "unable to set log file: " << e.what() << std::endl;
            }
         }
```
```suggestion
          if (auto log_file = local_conf()->log_file; !log_file.empty()) {
            try {
              logfile_set_ostream(log_file);
            } catch (const std::system_error& e) {
              std::cerr << "unable to set log file: " << e.what() << std::endl;
            }
         }
```
```suggestion
          if (auto log_file = local_conf()->log_file; !log_file.empty()) {
            try {
              logfile_set_ostream(log_file);
            } catch (const std::system_error& e) {
              std::cerr << "unable to set log file: " << e.what() << std::endl;
            }
         }
```
```suggestion
          if (auto log_file = local_conf()->log_file; !log_file.empty()) {
            try {
              logfile_set_ostream(log_file);
            } catch (const std::system_error& e) {
              std::cerr << "unable to set log file: " << e.what() << std::endl;
            }
         }
```
```suggestion
          if (auto log_file = local_conf()->log_file; !log_file.empty()) {
            try {
              logfile_set_ostream(log_file);
            } catch (const std::system_error& e) {
              std::cerr << "unable to set log file: " << e.what() << std::endl;
            }
         }
```

Reply:
Thanks @tchaikov i changed it like you suggested